index.js

```diff
@@ -60,48 +108,138 @@ app.post('/unregister', function(req, res) {
       res.sendStatus(404);
       return;
     }
-    client.del(token, function(err) {
-      res.sendStatus(200);
+    client.smembers(token, function(err, machines) {
+      for (var index = 0; index < machines.length; index++) {
+        client.del(machines[index]);
+      }
+      client.del(token, function(err) {
+        res.sendStatus(200);
```

Comment:
We should finish after removing both machines and token.
Maybe if we wrapped the redis API in promises all the code in index.js would get cleaner, we can do that in a followup.